src/idpi/grib_decoder.py

```diff
@@ -11,7 +11,22 @@
 import eccodes  # type: ignore
 import numpy as np
 import xarray as xr
-import yaml
+import yaml  # type: ignore
+
+
+DIM_MAP = {
+    "level": "z",
+    "perturbationNumber": "eps",
+    "step": "time",
+}
+VCOORD_TYPE = {
+    "generalVertical": ("model_level", -0.5),
+    "generalVerticalLayer": ("model_level", 0.0),
+    "hybrid": ("hybrid", 0.0),
+    "isobaricInPa": ("pressure", 0.0),
+    "surface": ("surface", 0.0),
+}
+_ifs_allowed = True
```

Comment:
why does this need to be a global variable?

Reply:
It's a flag that prevents ifs data from being loaded after cosmo data since conflicting definitions are being cached by eccodes. However, now that I think about it there should and equivalent comso_allowed flag since the reverse situation is also possible.

Reply:
two follow up questions here then:

Do we not have any way to clear the cache, in your warning you write to respawn the process. Is there no other control?
Now that this functionality requires state and access to constant dicts, would making it a class make sense?
2 can be an afterthought to this PR but we definitely need the flags going both directions

Reply:
There's an API to reset the grib context object but the issue is that it triggers a segmentation fault when called.

> Now that this functionality requires state and access to constant dicts, would making it a class make sense?

The class would still need global/static variables to provide the kind of functionality that we are looking for. I think it would bring little benefit.